flink-scala/src/main/scala/org/apache/flink/api/scala/GroupedDataSet.scala

```diff
@@ -356,6 +358,44 @@ class GroupedDataSet[T: ClassTag](
   }
 
   /**
+    * Applies a special case of a reduce transformation (maxBy) on a grouped {@link DataSet}.<br>
+    * The transformation consecutively calls a {@link ReduceFunction}
+    * until only a single element remains which is the result of the transformation.
+    * A ReduceFunction combines two elements into one new element of the same type.
+    *
+    * @param fields Keys taken into account for finding the minimum.
+    * @return A { @link ReduceOperator} representing the minimum.
+    */
+  def maxBy(fields: Array[Int]) : ReduceOperator[T]  = {
```

Comment:
Do not use `Array[Int]` as parameter. This is clumsy and not consistent with the remaining API. Use `fields: Int*` instead. `fields.toArray` converts to an Array.
This applies to all `maxBy`, `minBy` functions.

Do not use `ReduceOperator` as return type. This is a Java DataSet API class and must not be mix with the Scala API! Use `DataSet[T]` instead.